resources/assets/js/app/services/VehicleService.js

```diff
@@ -137,10 +138,25 @@ export const VehicleValidate = (data = {
     return result;
 };
 
+export const VehicleData = {
+    fetchBrand(name) {
+        return simpleRequest.get(`/api/v1/brands/${name}/`);
+    },
+
+    fetchModel(name, brandId) {
+        if (brandId > 0) {
+            return simpleRequest.get(`/api/v1/brands/${brandId}/models/${name}`);
+        } else {
+            return simpleRequest.get(`/api/v1/models/${name}/`);
```

Comment:
if brand wasn't selected from autocomplete, don't use autocomplete for models, also add filter Get param to filter collection by name